src/openloco/graphics/gfx.cpp

```diff
@@ -861,15 +867,46 @@ namespace openloco::gfx
     }
 
     // 0x004C5C69
+    // input:
+    // regs.ax: left
+    // regs.bx: top
+    // regs.dx: right
+    // regs.bp: bottom
     void set_dirty_blocks(int32_t left, int32_t top, int32_t right, int32_t bottom)
     {
-        registers regs;
-        regs.ax = left;
-        regs.bx = top;
-        regs.dx = right;
-        regs.bp = bottom;
-        call(0x004C5C69, regs);
+        if (left < 0)
+            left = 0;
+        if (top < 0)
+            top = 0;
+        if (right > _ui_width)
+            right = _ui_width;
+        if (bottom > _ui_height)
+            bottom = _ui_height;
+
+        if (left >= right)
+            return;
+        if (top >= bottom)
+            return;
+
+        right--;
+        bottom--;
+
+        left = left >> _dirty_blocks_columns;
```

Comment:
From this point onwards they are no longer representing a coordinate but a dirty block column I would prefer if we used a new variable therefore.